tedana/selection/select_comps.py

```diff
@@ -71,828 +46,298 @@ def selcomps(seldict, comptable, mmix, mask, ref_img, manacc, n_echos, t2s, s0,
-        LGR.warning('Extremely limited reliable BOLD signal space! '
-                    'Not filtering components beyond BOLD/non-BOLD guesses.')
-        midkfailed = True
-        min_acc = np.array([])
-        if len(group0) != 0:
-            # For extremes, building in a 20% tolerance
-            toacc_hi = np.setdiff1d(all_comps[utils.andb([fdist <= np.max(fdist[group0]),
-                                                          comptable['rho'] < F025,
-                                                          Vz > -2]) == 3],
-                                    np.union1d(group0, rej))
-            min_acc = np.union1d(group0, toacc_hi)
-            to_clf = np.setdiff1d(all_comps, np.union1d(min_acc, rej))
-        else:
-            toacc_hi = []
-            min_acc = []
-        diagstep_keys = ['Rejected components', 'Kappa-Rho cut point',
-                         'Kappa cut point', 'Rho cut point',
-                         'DBSCAN failed to converge',
-                         'Mid-Kappa failed (limited BOLD signal)',
-                         'Kappa-Rho guess',
-                         'min_acc', 'toacc_hi']
-        diagstep_vals = [list(rej), KRcut, Kcut, Rcut, dbscanfailed,
-                         midkfailed, list(KRguess), list(min_acc), list(toacc_hi)]
-        with open('csstepdata.json', 'w') as ofh:
-            json.dump(dict(zip(diagstep_keys, diagstep_vals)), ofh,
-                      indent=4, sort_keys=True, default=str)
-        return list(sorted(min_acc)), list(sorted(rej)), [], list(sorted(to_clf))
-
-    # Find additional components to reject based on Dice - doing this here
-    # since Dice is a little unstable, need to reference group0
-    rej_supp = []
-    dice_rej = False
-    if not dbscanfailed and len(rej) + len(group0) < 0.75 * len(all_comps):
-        dice_rej = True
-        temp = all_comps[dice_tbl[all_comps, 0] <= dice_tbl[all_comps, 1]]
-        rej_supp = np.setdiff1d(np.setdiff1d(np.union1d(rej, temp),
-                                             group0), group_n1)
-        comptable.loc[rej_supp, 'classification'] = 'rejected'
-        comptable.loc[rej_supp, 'rationale'] += 'Dice is bad;'
-        rej = np.union1d(rej, rej_supp)
-
-    # Temporal features
-    # larger is worse - spike
-    mmix_kurt_z = (mmix_kurt-mmix_kurt[group0].mean()) / mmix_kurt[group0].std()
-    # smaller is worse - drift
-    mmix_std_z = -1 * ((mmix_std-mmix_std[group0].mean()) / mmix_std[group0].std())
-    mmix_kurt_z_max = np.max([mmix_kurt_z, mmix_std_z], 0)
+    kappa_rate = ((np.max(comptable.loc[good_guess, 'kappa']) -
+                   np.min(comptable.loc[good_guess, 'kappa'])) /
+                  (np.max(comptable.loc[good_guess, 'variance explained']) -
+                   np.min(comptable.loc[good_guess, 'variance explained'])))
+    kappa_ratios = kappa_rate * comptable['variance explained'] / comptable['kappa']
+    varex_lb = stats.scoreatpercentile(
```

Comment:
`_lb` is lower bound, right ? Would you be opposed if I changed this to `_lower` and `_upper` ?